src/Core/Settings.h

```diff
@@ -774,6 +774,7 @@ class IColumn;
     M(Bool, input_format_tsv_detect_header, true, "Automatically detect header with names and types in TSV format", 0) \
     M(Bool, input_format_custom_detect_header, true, "Automatically detect header with names and types in CustomSeparated format", 0) \
     M(Bool, input_format_parquet_skip_columns_with_unsupported_types_in_schema_inference, false, "Skip columns with unsupported types while schema inference for format Parquet", 0) \
+    M(UInt64, input_format_parquet_max_block_size, 8192, "Max block size for parquet reader.", 0) \
```

Comment:
Hi, may I ask why the default value in here is 8192? @liuneng1994 @Avogar

Reply:
Probably it's to avoid using too much memory if rows are big, like in https://github.com/ClickHouse/ClickHouse/issues/45741 . E.g. if each row has tens of KBs of strings, block size of 100 K would eat gigabytes of memory per thread.

Yes, 8192 sounds too small for most cases.

Ideally we want a byte limit in addition to row count limit, but that seems nontrivial to implement. Reading happens column by column. All column readers have to stop after the same number of rows in order to form a valid block. I guess it'll have to be something like this:
```
while rows_so_far < 100 K && bytes_so_far < 10 MB:
    for each column:
        read 8192 rows for this column, appending to the current block
        bytes_so_far += size of the appended values
    rows_so_far += 8192
```
There are some difficulties:
 * AFAICT, arrow doesn't support appending to an existing column (or maybe I missed it); if we read into temporary memory and memcpy, this sounds like it'll mostly negate the performance gains from bigger blocks (or maybe not).
 * If we want to read columns in parallel in the future, this will become less useful because the inner "for each column" loop will also have thread scheduling overhead, so maybe we'd need some more complicated chunking strategy (I'm probably overcomplicating this).
 * I'm not sure how much effort we should spend on this vs writing our own parquet parser, which we should probably eventually do anyway.